### PR TITLE
fix bug #819: [BUG] [Service] [getPattern] Week info("firstDayOfWeek" &"weekendOfRange") should also follow CLDR data

### PR DESCRIPTION
**Pull request by huihuiw01**

### CODE REVIEW

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java [hunk 1]
```diff
@@ -563,6 +572,43 @@ public static String readZip(String fileName, String zipPath) {
...
     }
 
+    /**
+     * Get the first day of one Country, the implementation is according to https://www.unicode.org/reports/tr35/tr35-49/tr35-dates.html#Week_Data
+     *
+     * @param localeStr
+     * @param firstDayData
+     * @return
+     */
+    private static Integer firstDayExtract(String localeStr, Map<String, String> firstDayData){
```
Comment (tigershi) [inline g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java:582]: recommend the method name firstDayExtract to getFirstDayExtract

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java [hunk 2]
```diff
@@ -563,6 +572,43 @@ public static String readZip(String fileName, String zipPath) {
...
+    }
+
+    /**
+     * Get the weekend range of one Country, the implementation is according to https://www.unicode.org/reports/tr35/tr35-49/tr35-dates.html#Week_Data
+     *
+     * @param localeStr
+     * @param supplementalWeekData
+     * @return
+     */
+    private static List<Integer> weekendRangeExtract(String localeStr, Map<String, Object> supplementalWeekData){
```
Comment (tigershi) [inline g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java:597]: recommend the method name weekendRangeExtract to getWeekendRangeExtract

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java [hunk 3]
```diff
@@ -159,6 +159,38 @@ public String getDefaultRegionCode(String inputLocale, Map<String, String> regio
...
+	 * Get territory of a locale
+	 *
+	 * @param localeStr
+	 * @return
+	 */
+	public String getTerritory(String localeStr){
+		Locale locale = Locale.forLanguageTag(localeStr);
+		String territory = locale.getCountry();
+		if(territory == null || territory.isEmpty())//this is for supplement default region for locales like en, zh-Hans..
+			territory = getLanguageDefaultRegionMappings().get(localeStr);
```
Comment (Xiaochao8) [inline g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java:172]: I remember there is an existing method to get the default region of a locale. Could you check?

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java (hunk 2 above)
Comment (huihuiw01) [reply g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java]: Just for keep code style consistent with previous code, but it really doesn't conform the naming convention, so rename it to 'extractWeekendRange'.

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java (hunk 1 above)
Comment (huihuiw01) [reply g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java]: Just for keep code style consistent with previous code, but it really doesn't conform the naming convention, so rename it to 'extractFirstDay'.

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java (hunk 3 above)
Comment (huihuiw01) [reply g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java:172]: Confirmed with @chaoq0318, there is no such method.

Comment (Xiaochao8) [inline g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java:172]: How about this?
```suggestion
			territory = JSONUtils.getMapFromJson(PatternUtil.getRegionFromLib(localeStr)).get(ConstantsKeys.DEFAULT_REGION_CODE);
```

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java [hunk 4]
```diff
@@ -184,8 +193,8 @@ public static String readZip(String fileName, String zipPath) {
         dateMap.put(Constants.MONTH_FORMAT, monthsFormatMap);
         dateMap.put(Constants.MONTHS_STANDALONE, monthsStandaloneMap);
         dateMap.put(Constants.ERAS, erasMap);
-        dateMap.put(Constants.FIRST_DAY_OF_WEEK, 0);
-        dateMap.put(Constants.WEEKEND_RANGE, Arrays.asList(6, 0));
+        dateMap.put(Constants.FIRST_DAY_OF_WEEK, firstDayOfWeek);
+        dateMap.put(Constants.WEEKEND_RANGE, weekendRange);
```
Comment (Xiaochao8) [inline g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java:197]: Why add weekend range. How does this relate to the first day of week?

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java (hunk 3 above)
Comment (huihuiw01) [reply g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/LocaleDataUtils.java]: As I confirmed with @chaoq0318 and we synced yesterday, this method is not appropriate due to it use the extracted data and is used for calling by L2 API, so not appropriate to be used in the extracting process. 

I find another method which can be used here, and change code accordingly.

File: g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java (hunk 4 above)
Comment (huihuiw01) [reply g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/CLDRUtils.java:197]: As weekend range has the same bug, and need similar code change, so I fix them together.

I have asked @nannany00 include weekend range problem in the issue #879, and change PR title accordingly.

**Review by chaoq0318 — APPROVED** [review]

**Review by tigershi — APPROVED** [review]

**Review by tigershi — APPROVED** [review]

**Review by chaoq0318 — APPROVED** [review]

**Pull request merged**